actionview/test/template/url_helper_test.rb

```diff
@@ -15,6 +15,10 @@ class UrlHelperTest < ActiveSupport::TestCase
     get "/other" => "foo#other"
     get "/article/:id" => "foo#article", :as => :article
     get "/category/:category" => "foo#category"
+
+    scope :engine do
+      get "/" => "foo#bar"
```

Comment:
An even simpler way would be to ensure that we do not `chomp('/')` the `url_string` passed in by the user *if* the `request_uri` ends with `'/'`, otherwise we'd introduce the inequality which we later on check.
So as I mentioned in #29401 I think it would be easiest to change this line
`url_string.chomp!("/") if url_string.start_with?("/") && url_string != "/"`
to this line:
`url_string.chomp!("/") if url_string.start_with?("/") && !request_uri.end_with? "/"`